app/server/middleware/users.js

```diff
@@ -43,11 +44,15 @@ module.exports = {
             permissionRepository.getUserPermissionForRepo(username, repoId).then(permission => {
                 if (permission.permissive === 'admin' || permission.github === 'admin') {
                     users.forEach(user => {
-                        user.links = [{
+                        let editLink = new Link({
                             rel: 'edit-repo-permission',
-                            href: 'users/' + user.username + '/repos/' + repoId + '/permissions/{permission}',
-                            method: 'PUT'
-                        }];
+                            appMethod: 'users.editPermission',
+                            params: {
+                                id: repoId,
+                                username: user.username
+                            }
+                        });
+                        user.links = [editLink];
```

Comment:
It may make sense to have a middleware that initializes a link array, and then middleware like this that add links can append to that array.